src/eduid_userdb/orcid.py

```diff
@@ -2,687 +2,170 @@
-        """
-        :param value: Issuer
-        :type value: six.string_types
-        """
-        if not value or not isinstance(value, string_types):
-            raise UserDBValueError("Invalid 'iss': {!r}".format(value))
-        self._data['iss'] = value
-
-    # -----------------------------------------------------------------
-    @property
-    def sub(self):
-        """
-        Subject identifier
-
-        :return: subject id
-        :rtype: six.string_types
-        """
-        return self._data['sub']
-
-    @sub.setter
-    def sub(self, value):
-        """
-        :param value: Subject id
-        :type value: six.string_types
-        """
-        if not value or not isinstance(value, string_types):
-            raise UserDBValueError("Invalid 'sub': {!r}".format(value))
-        self._data['sub'] = value
-
-    # -----------------------------------------------------------------
-    @property
-    def aud(self):
-        """
-        Audience(s)
-
-        :return: audience list
-        :rtype: list
-        """
-        return self._data['aud']
-
-    @aud.setter
-    def aud(self, value):
-        """
-        :param value: audience list
-        :type value: list
-        """
-        if not isinstance(value, list):
-            raise UserDBValueError("Invalid 'aud': {!r}".format(value))
-        self._data['aud'] = value
-
-    # -----------------------------------------------------------------
-    @property
-    def exp(self):
-        """
-        Expiration time
-
-        :return: expiration time
-        :rtype: int
-        """
-        return self._data['exp']
-
-    @exp.setter
-    def exp(self, value):
-        """
-        :param value: expiration time
-        :type value: int
-        """
-        if not isinstance(value, int):
-            raise UserDBValueError("Invalid 'exp': {!r}".format(value))
-        self._data['exp'] = value
-
-    # -----------------------------------------------------------------
-    @property
-    def iat(self):
-        """
-        Expiration time
-
-        :return: expiration time
-        :rtype: int
-        """
-        return self._data['iat']
-
-    @iat.setter
-    def iat(self, value):
-        """
-        :param value: expiration time
-        :type value: int
-        """
-        if not isinstance(value, int):
-            raise UserDBValueError("Invalid 'iat': {!r}".format(value))
-        self._data['iat'] = value
-
-    # -----------------------------------------------------------------
-    @property
-    def nonce(self):
-        """
-        Nonce used to associate a Client session with an ID Token, and to mitigate replay attacks.
-
-        :return: nonce
-        :rtype: six.string_types
-        """
-        return self._data.get('nonce')
+    # Issuer identifier
+    iss: str
+    # Subject identifier
+    sub: str
+    # Audience(s)
+    aud: List[str]
+    # Expiration time
+    exp: int
+    # What is this?
+    iat: int
```

Comment:
iat is part of the OIDC ID token. "Time at which the JWT was issued. Its value is a JSON number representing the number of seconds from 1970-01-01T0:0:0Z as measured in UTC until the date/time."

Reply:
:+1: